sotodlib/toast/ops/sim_hwpss.py

```diff
@@ -97,6 +134,44 @@ def _exec(self, data, detectors=None, **kwargs):
             det_scale = unit_conversion(u.K, det_units)
 
             focalplane = obs.telescope.focalplane
+            fs = focalplane.sample_rate.to_value(u.Hz)
+
+            if self.drift_rate is not None and self.drift_rate != 0:
+                # Randomize the drift in a reproducible manner
+                counter1 = obs.session.uid
+                counter2 = self.realization
+                key1 = 683584
+                key2 = 476365
+                x = toast.rng.random(
+                    1,
+                    sampler="gaussian",
+                    key=(key1, key2),
+                    counter=(counter1, counter2),
+                )[0]
+                drift_rate = self.drift_rate * x
+                # Translate to actual drift
+                t = obs.shared[self.times].data
+                tmean = np.mean(t)  # Assumes data distribution by detector
+                drift = 1 + drift_rate * (t - tmean) / 3600
+            elif self.hwpss_random_drift and self.hwpss_drift_alpha is not None:
+                # Generate the HWPSS random drift common mode
+                counter1 = obs.session.uid
+                counter2 = self.realization
+                key1 = 683584
+                key2 = 476365
+                nsamp = obs.shared[self.times].data.size
+                w = toast.rng.random(
+                    nsamp,
+                    sampler="gaussian",
+                    key=(key1, key2),
+                    counter=(counter1, counter2),
+                ).array()
+                freqs = np.fft.rfftfreq(nsamp, 1/fs)
+                drift_psd = np.zeros_like(freqs)
+                drift_psd[1:] = np.abs(1/freqs[1:])**self.hwpss_drift_alpha
+                drift = np.fft.irfft(np.fft.rfft(w) * np.sqrt(drift_psd))      
```

Comment:
This produces zero-mean drift which is intuitively correct. However, since we apply the drift as a multiplicative factor, it should have unit mean (or median).